watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -1199,8 +1240,135 @@ def rule_visc_k_phase(b, p):
     def _radius_stokes_comp(self):
         add_object_reference(self, "radius_stokes_comp", self.params.radius_stokes_comp)
 
+    def _molar_volume_phase_comp(self):
+        add_object_reference(
+            self, "molar_volume_phase_comp", self.params.molar_volume_phase_comp
+        )
+
     def _diffus_phase_comp(self):
-        add_object_reference(self, "diffus_phase_comp", self.params.diffus_phase_comp)
+        # Retrieve two components sets from the molar_volume_data and diffusivity_data configurations
+        self.mv_dt_ind = Set(
+            initialize=[i[1] for i in self.params.config.molar_volume_data.keys()]
+        )
+        self.diffus_dt_ind = Set(
+            initialize=[i[1] for i in self.params.config.diffusivity_data.keys()]
+        )
+        if self.params.config.diffus_calculation == DiffusivityCalculation.none:
+            missing_diffus_ind = [
+                i
+                for i in (self.params.ion_set | self.params.solute_set)
+                if i not in self.diffus_dt_ind
+            ]
+            if not missing_diffus_ind == []:
+                _log.warning(
+                    "Diffusivity data is not provided for {}.".format(
+                        missing_diffus_ind
+                    )
+                )  # warning for components with no diffusivity_data entry.
+            self.diffus_phase_comp = Var(
+                self.params.phase_list,
+                self.diffus_dt_ind,
+                initialize=1e-9,
+                units=pyunits.m**2 * pyunits.s**-1,
+                doc="mass diffusivity of solute components",
+            )
+        elif (
+            self.params.config.diffus_calculation == DiffusivityCalculation.HaydukLaudie
+        ):
+            missing_diffus_ind = [
+                i
+                for i in (self.params.ion_set | self.params.solute_set)
```

Comment:
```suggestion
                for i in self.params.solute_set
```
Since PR #860 went through, I believe this can work instead.

Reply:
yes, I forgot to change this.  I'll push a commit.